README.adoc

```diff
@@ -97,21 +94,23 @@ include::finish/src/main/java/io/openliberty/guides/inventory/client/SystemClien
 ----
 
 The MicroProfile Rest Client feature automatically builds and generates an implementation for the `SystemClient` interface because the JAX-RS annotations in the template interface tell the RESTful service
-what to do. You don't have to worry about all of the boilerplate code, such as setting up a client class, connecting to the remote server, or invoking the correct URI with the correct parameters.
+what to do. Thus, you don't waste time on setting up the client and connecting with the remote service.
 
 When the `getProperties()` method is invoked, the `SystemClient` instance sends a GET request to the `<baseUrl>/properties` endpoint.
 
 The `@RegisterProvider` annotation tells the framework to register the provider classes to be used when the framework invokes the interface. You can add as many providers as necessary.
-In the `SystemClient` interface, add a response exception mapper as a provider to map the `404 NOT FOUND` response code with the `UnknownUrlException` exception.
+In the `SystemClient` interface, add a response exception mapper as a provider to map the `404` response code with the `UnknownUrlException` exception.
 
 // =================================================================================================
 // Handling exceptions through ResponseExceptionMappers
 // =================================================================================================
 === Handling exceptions through ResponseExceptionMappers
 
-In the `SystemClient` interface, the `getProperties()` method might throw an `UnknownUrlException` exception if the response from the `system` service is `404 NOT FOUND`.
-The interface maps to what you want to invoke on the remote service. Then, the `@RegisterProvider(UnknownUrlExceptionMapper.class)` annotation registers the response exception mapper.
-Exception mappers map various response codes from the remote service to throwable exceptions.
+Error handling is an important step to ensure the application can fail safely. If there is an error response when invoking the remote service, for instance, the service returns a `404` response code,
+you need to handle it. First, you need to define an exception, and map the exception with the error response code. Then, register the exception mapper in the client interface.
+
+Look at the client interface from the previous section, the `@RegisterProvider(UnknownUrlExceptionMapper.class)` annotation registers the `UnknownUrlExceptionMapper` response exception mapper.
```

Comment:
"...previous section. The `@RegisterProvider(UnknownUrlExceptionMapper.class)` annotation..."

It might be even better if we can omit "from the previous section" because the guide and how it's ordered could change in the future.